### ISSUE TITLE
Main navigation not fully keyboard accessible

### ISSUE THREAD

**Issue by schalkneethling** [mdn/developer-portal#1312]
**Describe the bug**
Some of the top-level menu entries for the main navigation are skipped when using the keyboard to navigate around the page.

**To Reproduce**
Best to test this in Chrome or Safari if on a Mac as the Firefox needs some prodding to get it to behave the same way :)

1. Go to the landing page
2. Start pressing tab
3. Focus will first be on the `x` close icon for the task completion survey notice
4. Keep pressing tab. You will notice that focus for `Products & Technologies` and `Communities` is skipped

**Expected behavior**

All menu items should get keyboard focus in a predictable order. Also, once a menu item with children receive focus, pressing enter should open up the submenu

**Screenshots**

N/A

**Desktop (please complete the following information):**
 - OS: All
 - Browser All
 - Version All

**Smartphone (please complete the following information):**
N/A

**Additional context**
The problem is caused by the top-level menu entries marked up as `span` elements as opposed to `a` elements. It is also important to note that the `a` elements will need a valid `href`.

For example this,

```
<span class="mzp-c-menu-title" aria-haspopup="true" aria-controls="mzp-c-menu-panel-topics" aria-expanded="false">Products &amp; Technologies</span>
```

Should be changed to:

```
<a href="https://developer.mozilla.com/" class="mzp-c-menu-title" aria-haspopup="true" aria-controls="mzp-c-menu-panel-topics" aria-expanded="false">Products &amp; Technologies</a>
```

**Comment by stevejalim** [comment]
Note that this is the behaviour on the current design, too, so it's not technically a redesign regression, but I definitely agree that it's worth sorting out
